index.js

```diff
@@ -142,59 +154,177 @@ function updateSchemesAndHost(req) {
+  });
+}
+
+/**
+ * TODO
+ *
+ * 1. Rename the parameter names
+ * (this will require better global variable naming to allow re-assigning properly)
+ *
+ * 2. the `aPath` is ignored only if it's `undefined`,
+ * but if it's set to an empty string `''`, then the tests fail.
+ * I think we should be checking for falsy values, not only `undefined` ones.
+ *
+ * 3. (Breaking) Use object for optional parameters:
+ * https://github.com/mpashkovskiy/express-oas-generator/issues/35
+ *
+ */
+/**
+ * @warn it's preferred that you use `injectResponseMiddleware`,
+ * `injectRequestMiddleware` and `serveApiDocs` **individually**
+ * and not directly from this `init` function,
+ * because we need `injectRequestMiddleware` to be placed as the
+ * very last middleware and we cannot guarantee this here,
+ * since we're only using an arbitrary setTimeout of `1000` ms.
+ *
+ * See
+ * https://github.com/mpashkovskiy/express-oas-generator/pull/32#issuecomment-546807216
+ *
+ * @description initialize the `express-oas-generator`.
+ *
+ * This will apply both `injectResponseMiddleware` and `injectRequestMiddleware`
+ * and also will call `serveApiDocs`.
+ *
+ * @param {Express} aApp - the express app
+ * @param {*} [aPredefinedSpec=undefined]
+ * @param {string|undefined} [aPath=undefined] where to write the openAPI specification to.
+ * Specify this to create the openAPI specification file.
+ * @param {number} [aWriteInterval=10000] how often to write the openAPI specification to file
+ * @param {string} [aApiDocsPath=api-docs] where to serve the openAPI docs. Defaults to `api-docs`
+ */
+function init(aApp, aPredefinedSpec = undefined, aPath = undefined, aWriteInterval = 1000 * 10, aApiDocsPath = 'api-docs') {
+  /**
+   * TODO - shouldn't `predefinedSpec` be assigned @ `serveApiDocs`?
+   *
+   * I don't know if the `predefinedSpec` is used anywhere for `injectResponseMiddleware`
+   * and before `injectRequestMiddleware` is called
+   */
+  predefinedSpec = aPredefinedSpec;
+
+  injectResponseMiddleware(aApp, { pathToOutputFile: aPath, writeIntervalMs: aWriteInterval });
+
+  /**
+   * make sure we list routes after they are configured
+   *
+   * TODO - this (setTimeout) is error-prone.
+   * See https://github.com/mpashkovskiy/express-oas-generator/pull/32#issuecomment-546807216
+   *
+   * There could be some heavylifing initialization that takes
+   * more than a second and in those cases the requests processing middleware
+   * wouldn't be the last one.
+   */
+  setTimeout(() => {
+    injectRequestMiddleware();
+    serveApiDocs({ path: aApiDocsPath, predefinedSpec });
   }, 1000);
-};
+}
 
-module.exports.getSpec = () => {
+const getSpec = () => {
   return patchSpec(predefinedSpec);
 };
 
-module.exports.setPackageInfoPath = pkgInfoPath => {
+const setPackageInfoPath = pkgInfoPath => {
   packageJsonPath = `${process.cwd()}/${pkgInfoPath}/package.json`;
 };
+
+module.exports = {
+  injectResponseMiddleware,
+  injectRequestMiddleware,
+  serveApiDocs,
```

Comment:
`serveApiDocs()` is internal (private) method, shold not be exported

Reply:
Got it. 05b1e6a